pandas/core/indexing.py

```diff
@@ -1869,6 +1869,12 @@ def _setitem_single_column(self, loc: int, value, plane_indexer):
         """
         pi = plane_indexer
 
+        if not getattr(self.obj._mgr, "blocks", False):
```

Comment:
can you use explicit isinstance; e.g. at first glance empty blocks would go through here

Reply:
Ah, I should have used `hasattr` instead of `getattr` here (which would avoid that empty block case). 

`pandas.core.indexing` currently doesn't import from internals, which I suppose is the reason that I used this implicit check instead of an explicit isinstance (having it as a property on the object, which I think was added in some other (open) PR, might also help for this case)